### ISSUE TITLE
Do we want to invest in this?

### ISSUE THREAD

**Issue by matthew-andrews** [Financial-Times/next-component-tools#4]
Perhaps my views are a little tainted because when I came to use next-component-tools for the first time they didn't work (and continue not to work) and I am now effectively locked out from being able to work on half of Next.

Origami Build Tools do a very similar job to NCT.  The development workflow isn't perfect (uses Mustache internally rather than Handlebars, OBT's watch task [could be optimised](https://github.com/Financial-Times/origami-build-tools/issues/259)) but they offer a number of very compelling benefits:-
- it works on my laptop
- extensive test suite
- it's a team of people's job to fix bugs in it
- it's a team of people's job to help people install it
- it's *substantially* faster to run than NCT.  (Waiting minutes for `nct demo` to get going is not fun)

I'd really, really, really rather not spend another week on build tools but I am very close to deleting this repo and making every component work with Origami Build Tools again.  Please convince me that this is a bad idea.

@wheresrhys @commuterjoy

**Comment by wheresrhys** [comment]
I arranged a meeting on friday (hopefully in your calendar) to discuss. I agree in principle with you that it's best to try to get good ideas from here into obt (the meeting is partly to evaluate, with people who've used this tool, which ideas are worth pursuing) and also, despite it being quite fun to hack together this proof of concept, don't relish the prospect of maintaining this module.

But realistically we will always need a tool for QA that can render real component templates in a variety of states. Origami doesn't have this requirement as its templates are not meant as application code, so a template change is a major release and they don't have to worry so much about template bugs finding their way into lots of production applications at once. Templating has been an intractable issue for origami for some time so I made this tool because obt is not going to solve the templating QA problem for us in the foreseeable future. I also - and I know you disagree on this point - think not having to maintain demo templates in parallel with the real handlebars templates is a big win; removes a manual task which saves time and minimises the risk of things falling out of step.

For the time being I can change nct to look for a `next-demos` directory, that way you can still have obt demos in the same module and this won't be a blocker for you. The speed issue is a weird one. All the way through developing it I had no speed issues but have noticed it's started to crawl. Obviously it's a problem with my implementation, but it doesn't mean a demoer that runs a server consuming our real templates with real flags, helpers and data isn't a beneficial thing.

**Comment by matthew-andrews** [comment]
We can chat more on Friday but in the mean time I can't do any work so please can you fix https://github.com/Financial-Times/next-component-tools/issues/5.

> think not having to maintain demo templates in parallel

I thought your tool that generated mustache templates from handlebar ones was a good workaround for this that solve this:-

> removes a manual task which saves time and minimises the risk of things falling out of step.

**Comment by matthew-andrews** [comment]
Here's what I tried tonight:-

- Upgrade to Yosemite
- Remove nodenv (just using Node v0.12.2) directly
- Try OBT 3.0.0-0

(Still using Sass 3.4.13)

Same symptoms as before :disappointed:.

Probably not ideal that next-header will [now only work on my computer](https://github.com/Financial-Times/next-header/blob/master/Makefile#L16) :laughing:

**Comment by AlbertoElias** [comment]
What's the outcome of this? Does Next need a separate tool for building demos? I've noticed Next components build demos in many different ways and doesn't seem to standardise on one or the other. Shouldn't a decision to standardise on that be made (if it hasn't) before building a new tool?

I understand you probably do need it like @wheresrhys to test the templates you serve, but maybe not to actually build the demos if you decide to standardise on the Origami way of building demos which would massively help for all the components that will be merged into Origami. 

Might be good to come to the Origami spec meeting to talk about this today. We can also discuss @keirog's current solution. cc @matthew-andrews
